src/main/java/com/nti/socialmediaappcore/controller/AuthController.java

```diff
@@ -0,0 +1,130 @@
+package com.nti.socialmediaappcore.controller;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.Valid;
+
+import com.nti.socialmediaappcore.payload.request.LoginRequest;
+import com.nti.socialmediaappcore.payload.request.SignupRequest;
+import com.nti.socialmediaappcore.payload.response.JwtResponse;
+import com.nti.socialmediaappcore.payload.response.MessageResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.nti.socialmediaappcore.jwt.JwtUtils;
+import com.nti.socialmediaappcore.model.ERole;
+import com.nti.socialmediaappcore.model.Role;
+import com.nti.socialmediaappcore.model.User;
+import com.nti.socialmediaappcore.repository.RoleRepository;
+import com.nti.socialmediaappcore.repository.UserRepository;
+import com.nti.socialmediaappcore.service.UserDetailsImpl;
+
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    RoleRepository roleRepository;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Autowired
+    JwtUtils jwtUtils;
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
```

Comment:
Rename JwtResponse class to e.g. UserDetailsDTO (sth more descriptive). Remove **token** and **type** fields  Add other fields to this class if/when needed - it should contain user-private data, stored settings, preferences etc. Generally, any user information that would be useful after login. 

Generated jwt return in the response header:  https://stackoverflow.com/a/50932596

```suggestion
    public UserDetailsDTO authenticateUser(@Valid @RequestBody LoginRequest loginRequest, HttpServletResponse response) {
    
    ...
    
    response.setHeader("Token", jwt);
    return userDetailsDTO;
```